jupyter_console/interactiveshell.py

```diff
@@ -220,6 +224,43 @@ def handle_execute_reply(self, msg_id, timeout=None):
 
             self.execution_count = int(content["execution_count"] + 1)
 
+    def handle_is_complete_reply(self, msg_id, timeout=None):
+        """
+        `complete` - code is ready to be executed
+        `incomplete` - code should prompt for another line
+        `invalid` - code will typically be sent for execution, so that the user sees the error soonest.
+        `unknown` - if the kernel is not able to determine this. The
+                    frontend should also handle the kernel not replying promptly. It
+                    may default to sending the code for execution, or it may implement
+                    simple fallback heuristics for whether to execute the code
+                    (e.g. execute after a blank line).
+
+        self._source is the code in question.
+        return more, indent
+        """
+        ## Get the is_complete response:
+        msg = None
+        while self.client.is_alive():
+            try:
+                msg = self.client.shell_channel.get_msg(block=False, timeout=timeout)
+            except Empty:
+                continue
+            if msg["parent_header"].get("msg_id", None) == msg_id:
+                break
+        ## Handle response:
+        status = msg["content"].get("status", None)
+        indent = msg["content"].get("indent", "")
+        if status == "complete":
+            return False, indent
+        elif status == "incomplete":
+            return True, indent
+        elif status == "invalid":
+            raise SyntaxError()
```

Comment:
Since we're only doing this so it can be caught and turned into `more = False`, maybe it should just return `False, ''` directly?